finetune/eval/if_eval/start_end.py

```diff
@@ -0,0 +1,24 @@
+import random
+
+from finetune.eval.if_eval.rule import IFEvalRule, RuleId
+
+ENDS_WITH_PHRASES = [
+    "Let me know if you have additional questions.",
+    "Is there anything else I can help with?",
+    "Hope that helps.",
+]
+
+
+class EndsWithRule(IFEvalRule):
+    """Rule that enforces the response ends with an exact phrase."""
+
+    def __init__(self):
+        super().__init__(rule_id=RuleId.ENDS_WITH)
+        self.phrase = random.choice(ENDS_WITH_PHRASES)
+
+    def get_prompt(self, index: int = -1) -> str:
+        return f'End your response with the exact phrase "{self.phrase}"'
+
+    def matches(self, text: str, index: int = -1) -> bool:
+        # Check matches while ignoring casing.
+        return text.lower().endswith(self.phrase.lower())
```

Comment:
Should we rstrip it in case a newline was added?

Reply:
I think that would mean you fail the rule. I've added a test case that this fails but could be convinced to change it.